truetype/truetype.go

```diff
@@ -201,13 +201,18 @@ func (f *Font) parseCmap() error {
 		return err
 	}
 	offset = int(u32(f.cmap, offset+4))
-	if offset <= 0 || offset > len(f.cmap) {
+	cmapLength := len(f.cmap)
+	if offset <= 0 || offset+2 > cmapLength {
```

Comment:
It probably won't matter in practice, but I'd still rather avoid overflow issues:

if offset <= 0 || offset > cmapLength-2 {

Ditto below.